keystone_performance/load_test.py

```diff
@@ -222,47 +294,54 @@ def _print(self):
 
 
 class Request(object):
-    def __init__(self, agent, request_gatherer, args, on_complete=None):
+    def __init__(self, agent, request_gatherer, args, user_info, on_complete=None):
         self._agent = agent
         self._request_gatherer = request_gatherer
         self._args = args
+        self._user_info = user_info
         self._on_complete = on_complete
 
         self._request_no = 0
         self._done = False
 
-        if self._args.user_domain_id:
-            user_domain_info = {'id': self._args.user_domain_id}
-        else:
-            user_domain_info = {'name': self._args.user_domain_name}
-        if self._args.project_id:
-            project_info = {'id': self._args.project_id}
-        else:
-            project_info = {'name': self._args.project_name}
-        if self._args.project_domain_id:
-            project_info['domain'] = {'id': self._args.project_domain_id}
+    def start(self):
+        user_info = self._user_info.get_random_user_info()
+
+        project_info = None
+        if 'project_id' in user_info:
```

Comment:
This is confusing because we can see that user_info is only going to have username and domain_id.